SnowflakeToS3Operator

JavierLopezT opened the pull request:
New transfer operator, Snowflake To S3

JavierLTPromofarma committed:
SnowflakeToS3Operator

JavierLTPromofarma committed:
SnowflakeToS3Operator

JavierLTPromofarma committed:
SnowflakeToS3Operator

JavierLTPromofarma committed:
SnowflakeToS3Operator

JavierLTPromofarma committed:
SnowflakeToS3Operator

JavierLTPromofarma committed:
SnowflakeToS3Operator

JavierLTPromofarma committed:
fix

JavierLTPromofarma committed:
fix

JavierLTPromofarma committed:
providers

JavierLTPromofarma committed:
providers

turbaszek commented on airflow/providers/amazon/aws/transfers/snowflake_to_s3.py:
```diff
@@ -0,0 +1,82 @@
+"""
```
Missing Apache license header

turbaszek commented on airflow/providers/amazon/aws/transfers/snowflake_to_s3.py:
```diff
@@ -0,0 +1,82 @@
+        snowflake_conn_id: str = "snowflake_default",
+        unload_options: Optional[str] = None,
+        autocommit: bool = True,
+        *args, **kwargs) -> None:
```
```suggestion
        self,
        *,
        s3_bucket: str,
        s3_key: str,
        file_format: str,
        schema: Optional[str] = None,
        table: Optional[str] = None,
        sql: Optional[str] = None,
        snowflake_conn_id: str = "snowflake_default",
        unload_options: Optional[str] = None,
        autocommit: bool = True,
        **kwargs) -> None:
```

JavierLopezT committed:
Update airflow/providers/amazon/aws/transfers/snowflake_to_s3.py

Co-authored-by: Tomek Urbaszek <turbaszek@gmail.com>

JavierLopezT committed:
Update airflow/providers/amazon/aws/transfers/snowflake_to_s3.py

Co-authored-by: Tomek Urbaszek <turbaszek@gmail.com>

JavierLTPromofarma committed:
tests, apache header, new argument, change in unload_options

JavierLTPromofarma committed:
tests, apache header, new argument, change in unload_options

JavierLTPromofarma committed:
fixes

JavierLTPromofarma committed:
fixes

JavierLTPromofarma committed:
fixes

JavierLTPromofarma committed:
fixes

JavierLTPromofarma committed:
fixes

JavierLTPromofarma committed:
fixes

JavierLTPromofarma committed:
color

JavierLTPromofarma committed:
color

turbaszek commented on airflow/providers/amazon/aws/transfers/snowflake_to_s3.py:
```diff
@@ -0,0 +1,111 @@
+        ``type = csv
+         field_delimiter = ','
+         skip_header = 1
+         compression=None``
```
Missing indent

turbaszek commented on airflow/providers/amazon/aws/transfers/snowflake_to_s3.py:
```diff
@@ -0,0 +1,111 @@
+    :param snowflake_conn_id: reference to a specific snowflake database
+    :type snowflake_conn_id: str
+    :param unload_options: reference to a list of UNLOAD options (SINGLE, MAX_FILE_SIZE,
+    OVERWRITE etc). Each element of the list has to be a string
```
Missing indent

turbaszek commented on airflow/providers/amazon/aws/transfers/snowflake_to_s3.py:
```diff
@@ -0,0 +1,111 @@
+                    FILE_FORMAT = ({self.file_format})
+                    {unload_options}
+                    HEADER = {self.include_header};
+                    """
```
Do we need this `\n\t\t\t` because the `unload_query` is indented? Should we be able to dedent it?

JavierLopezT replied:
To be honest, I just copied it from the existing RedshiftToS3Operator. I don't know if we should maintain this or modifying. What do you think?

turbaszek replied:
I think if we can improve the code and remove this strange string then it may be worth doing it

JavierLopezT replied:
I was able to remove two \t characters. Otherwise, the query in the log is unintended.

JavierLTPromofarma committed:
indent

JavierLTPromofarma committed:
indent

JavierLTPromofarma committed:
simplify string

JavierLTPromofarma committed:
simplify string

JavierLTPromofarma committed:
simplify string test

JavierLTPromofarma committed:
simplify string test

JavierLTPromofarma committed:
fixes

JavierLTPromofarma committed:
fixes

JavierLTPromofarma committed:
doc fixes

JavierLTPromofarma committed:
doc fixes

JavierLTPromofarma committed:
adding hardcoded to spelling wordlist

JavierLTPromofarma committed:
adding hardcoded to spelling wordlist

JavierLTPromofarma committed:
fix test

JavierLTPromofarma committed:
fix test

RosterIn commented on airflow/providers/amazon/aws/transfers/snowflake_to_s3.py:
```diff
@@ -0,0 +1,105 @@
+
+class SnowflakeToS3Operator(BaseOperator):
+    """
+    UNLOAD a query or a table from Snowflake to S3
```
I think this operator needs further documentation.
Some users might be surprised that there is no `aws_conn` here and the `aws_access_key_id`, `aws_secret_access_key` are defined in snowflake connection. I think this is not trivial.

JavierLopezT replied:
I have added some docs as well as another possibility to unload. What do you think?

RosterIn commented on tests/providers/amazon/aws/transfers/test_snowflake_to_s3.py:
```diff
@@ -0,0 +1,69 @@
+            'OVERWRITE = TRUE',
+        ]
+        file_format = "file_format"
+        sql = None
```
The operator has two modes one when sql param is provided and one when it's not. Shouldn't be two test cases?

feluelle commented on airflow/providers/amazon/aws/transfers/snowflake_to_s3.py:
```diff
@@ -0,0 +1,105 @@
+        unload_options = '\n\t'.join(self.unload_options)
+
+        if not self.sql:
+            self.sql = self.schema + '.' + self.table
```
```suggestion
```
I think this makes it unnecessary "complex".

feluelle commented on airflow/providers/amazon/aws/transfers/snowflake_to_s3.py:
```diff
@@ -0,0 +1,105 @@
+        file_format: str,
+        schema: Optional[str] = None,
+        table: Optional[str] = None,
+        sql: Optional[str] = None,
```
```suggestion
        table_or_query: Optional[str] = None,
```
I would suggest to keep it rather simple and allow passing table or query to one variable and just pass it to the `COPY` string. Otherwise you would have to do more validation on this - so that either schema and table or a query was passed.

Ref: https://docs.snowflake.com/en/sql-reference/sql/copy-into-location.html#syntax

JavierLopezT replied:
Done it.

JavierLTPromofarma committed:
changes

JavierLTPromofarma committed:
changes

JavierLTPromofarma committed:
doc fix

JavierLTPromofarma committed:
doc fix

JavierLTPromofarma committed:
doc fix

JavierLTPromofarma committed:
doc fix

JavierLTPromofarma committed:
doc fix

JavierLTPromofarma committed:
doc fix

JavierLTPromofarma committed:
doc fix

JavierLTPromofarma committed:
doc fix

JavierLTPromofarma committed:
static checks

JavierLTPromofarma committed:
static checks

JavierLTPromofarma committed:
update code

JavierLTPromofarma committed:
update code

JavierLTPromofarma committed:
fixes

JavierLTPromofarma committed:
fixes

JavierLTPromofarma committed:
typo

JavierLTPromofarma committed:
typo

JavierLTPromofarma committed:
typo

RosterIn commented on airflow/providers/amazon/aws/transfers/snowflake_to_s3.py:
```diff
@@ -0,0 +1,150 @@
+        or hardcoded one like ``type = csv field_delimiter = ',' skip_header = 1``
+    :type file_format: str
+    :param query_or_table: query or full table to unload. If table, it must include the schema like
+        ``schema.table`
```
```suggestion
        `schema.table`
```

JavierLopezT committed:
Update airflow/providers/amazon/aws/transfers/snowflake_to_s3.py

Co-authored-by: RosterIn <48057736+RosterIn@users.noreply.github.com>

JavierLopezT committed:
Update airflow/providers/amazon/aws/transfers/snowflake_to_s3.py

Co-authored-by: RosterIn <48057736+RosterIn@users.noreply.github.com>

feluelle commented on airflow/providers/amazon/aws/transfers/snowflake_to_s3.py:
```diff
@@ -0,0 +1,150 @@
+        `schema.table`
+    :type query_or_table: str
+    :param snowflake_conn_id: reference to a specific snowflake database
+    :type snowflake_conn_id: str`
```
```suggestion
    :type snowflake_conn_id: str
```

feluelle commented on airflow/providers/amazon/aws/transfers/snowflake_to_s3.py:
```diff
@@ -0,0 +1,150 @@
+    :param stage: Copy the data into a Snowflake Stage. This allows you to use the method
+        'Configuring AWS IAM User Credentials' for unloading data to s3. If this is true,
+        s3_bucket and file_format are not necessaries.
+    :type stage: bool
```
```suggestion
    :param use_stage: Copy the data into a Snowflake Stage. This allows you to use the method
        'Configuring AWS IAM User Credentials' for unloading data to s3. If this is true,
        s3_bucket and file_format are not necessaries.
    :type use_stage: bool
```
WDYT? If yes, then obviously you have to change the references. (I do not comment on them. :))

JavierLopezT replied:
Actually this is a mistake, stage is a string. I have already changed it. Thanks!

feluelle reviewed (commented):
LGTM, only one suggestion :) 👍

JavierLopezT committed:
Update airflow/providers/amazon/aws/transfers/snowflake_to_s3.py

Co-authored-by: Felix Uellendall <feluelle@users.noreply.github.com>

JavierLopezT committed:
Update airflow/providers/amazon/aws/transfers/snowflake_to_s3.py

Co-authored-by: Felix Uellendall <feluelle@users.noreply.github.com>

JavierLTPromofarma committed:
changes

JavierLTPromofarma committed:
changes

JavierLTPromofarma committed:
typo

JavierLTPromofarma committed:
typo

JavierLopezT committed:
Merge branch 'master' into snowflake_to_s3

JavierLopezT committed:
Merge branch 'master' into snowflake_to_s3

ashb committed:
Make v1/config endpoint respect webserver expose_config setting (#14020)

ashb committed:
Remove use of repeated constant in AirflowConfigParser (#14023)

Small refactor that doesn't change any functionality, but does make
future testing/refactoring easier.

mik-laj committed:
Fix four bugs in StackdriverTaskHandler (#13784)

mik-laj committed:
Simplify Kerberos network setup (#13999)

stubs committed:
Fix typos in Upgrade Check Doc (#14035)

kaxil committed:
Disable row level locking for Mariadb and MySQL <8 (#14031)

closes #11899
closes #13668

This PR disable row-level locking for MySQL variants that do not support skip_locked and no_wait -- MySQL < 8 and MariaDB

jhtimmins committed:
Bugfix: Fix permissions to triggering only specific DAGs  (#13922)

From 1.10.x -> 2.0, the required permissions to trigger a dag have changed from DAG.can_edit to DAG.can_read + DagRun.can_create. Since the Viewer role has DAG.can_read by default, it isn't possible to give a Viewer trigger access to a single DAG without giving access to all DAGs.

This fixes that discrepancy by making the trigger requirement DAG.can_edit + DagRun.can_create. Now, to trigger a DAG, a viewer will need to be given both permissions, as neither is with the Viewer role by default.

This PR also hides the Trigger/Refresh buttons on the home page if the user doesn't have permission to perform those actions.

closes: #13891
related: #13891

mik-laj committed:
Support google-cloud-logging` >=2.0.0 (#13801)

Commit:
Add retryer to SFTP hook connection (#13065)

ryanahamilton committed:
Utilize util method to yield versioned doc link (#14047)

Commit:
Fix critical CeleryKubernetesExecutor bug (#13247)

closes https://github.com/apache/airflow/issues/13263

XD-DENG committed:
Clean-up JS code in UI templates (#14019)

- Use template literals instead of '+' for forming strings, when applicable
- remove unused variables (gantt.html)
- remove unused function arguments, when applicable

Commit:
Add Apache Beam operators (#12814)

eladkal committed:
Add WeekDayBranchOperator (#13997)

potiuk committed:
Prepare to release a new wave of providers. (#14013)

For the regular providers, Vast majority is in `1.0.1` version and it is
only documentation update - but this way we will have a consistent set
of documentation (including commit history) as well as when we release
in PyPI, the READMES will be much smaller and link to the documentation.

We have two new providers (version 1.0.0):

* neo4j
* apache.beam

There are few providers with changes:

Breaking changes (2.0.0)

* google
* slack

Feature changes (1.1.0):

* amazon
* exasol
* http
* microsoft.azure
* openfaas
* sftp
* snowflake
* ssh

There were also few providers with 'real' bugfixes (1.0.1):

* apache.hive
* cncf.kubernetes
* docker
* elasticsearch
* exasol
* mysql
* openfaas
* papermill
* presto
* sendgrid
* sqlite

The ''backport packages" documentation is prepared only for those
providers that had actual bugfix/features/breaking changes:

```
amazon apache.hive cncf.kubernetes docker elasticsearch exasol google
http microsoft.azure mysql openfaas papermill presto sendgrid sftp
slack snowflake sqlite ssh
```

Only those will be generated with `2021.2.5` calver version.

jhtimmins committed:
Make the role assigned to anonymous users customizable (#14042)

Fixes the issue wherein regardless of what role anonymous users are assigned (via the `AUTH_ROLE_PUBLIC` env var), they can't see any DAGs.

Current behavior causes:
Anonymous users are handled as a special case by Airflow's DAG-related security methods (`.has_access()` and `.get_accessible_dags()`). Rather than checking the `AUTH_ROLE_PUBLIC` value to check for role permissions, the methods reject access to view or edit any DAGs.

Changes in this PR:
Rather than hardcoding permission rules inside the security methods, this change checks the `AUTH_ROLE_PUBLIC` value and gives anonymous users all permissions linked to the designated role. 

**This places security in the hands of the Airflow users. If the value is set to `Admin`, anonymous users will have full admin functionality.**

This also changes how the `Public` role is created. Currently, the `Public` role is created automatically by Flask App Builder. This PR explicitly declares `Public` as a default role with no permissions in `security.py`. This change makes it easier to test.

closes: #13340

kaxil committed:
Retry critical methods in Scheduler loop in case of OperationalError (#14032)

In the case of OperationalError (caused deadlocks, network blips), the scheduler will now retry those methods 3 times.

closes #11899
closes #13668

kaxil committed:
Fix broken SLA Mechanism (#14056)

closes https://github.com/apache/airflow/issues/14050

We were not de-serializing `BaseOperator.sla` properly, hence
we were returning float instead of `timedelta` object.

Example: 100.0 instead of timedelta(seconds=100)

And because we had a check in _manage_sla in `SchedulerJob` and `DagFileProcessor`,
we were skipping SLA.

SchedulerJob:
https://github.com/apache/airflow/blob/88bdcfa0df5bcb4c489486e05826544b428c8f43/airflow/jobs/scheduler_job.py#L1766-L1768

DagFileProcessor:
https://github.com/apache/airflow/blob/88bdcfa0df5bcb4c489486e05826544b428c8f43/airflow/jobs/scheduler_job.py#L395-L397

turbaszek committed:
Remove unused 'context' variable in task_instance.py (#14049)

VBhojawala committed:
Added json_render method to separate filtering from view (#14024)

potiuk committed:
Fixes to release process after releasing 2nd wave of providers (#14059)

kaxil committed:
Bugfix: Scheduler fails if task is removed at runtime (#14057)

closes https://github.com/apache/airflow/issues/13464

jhtimmins committed:
Correctly capture debug logs in plugin tests. (#14058)

This fixes the test test_should_load_plugins_from_property, which is currently quarantined as a "Heisentest".

Current behavior:
The test currently fails because the records that it expects to find in the logger are not present.

Cause:
While the test sets the logger as "DEBUG", it doesn't specify which logger to update. Python loggers are namespaced (typically based on the current file's path), but this has to be defined explicitly. In the absence of a specified logger, any attempts to lookup will return the BaseLogger instance.

The test is therefore updating the log level for the base logger, but when the test runs, the plugins_manager.py file defines a namespaced logger log = logging.getLogger(__name__) used throughout the file. Since a different logger is used, the original log level, in this case INFO, is used. INFO is a higher level than DEBUG, so the calls to log.debug() get filtered out, and when the test looks for log records it finds an empty list.

Fix:
Just specify which logger to update when modifying the log level in the test.

cccs-seb committed:
Add return to PythonVirtualenvOperator's execute method (#14061)

ashb committed:
Update to Pytest 6.0 (#14065)

And pytest 6 removed a class that the rerunfailures plugin was using, so
we have to upgrade that too.

kaxil committed:
Remove permissions to read Configurations for User and Viewer roles (#14067)

Only `Admin` or `Op` roles should have permissions to view Configurations.

Previously, Users with `User` or `Viewer` role were able to get/view configurations using
the REST API or in the Webserver. From Airflow 2.0.1, only users with `Admin` or `Op` role would be able
to get/view Configurations.

varunvora committed:
Fix DAGs mount path in Kubernetes worker pod when gitSync is enabled (#13826)

* Update pod-template-file.kubernetes-helm-yaml

* Fix ssh-key access issue

This change allows dags.gitSync.containerName to read ssh-key from file system.
Similar to this https://github.com/varunvora/airflow/blob/ce0e6280d2ea39838e9f0617625cd07a757c3461/chart/templates/scheduler/scheduler-deployment.yaml#L92
It solves https://github.com/apache/airflow/issues/13680 issue for private repositories.

Co-authored-by: Denis Krivenko <36439732+dnskr@users.noreply.github.com>

leahecole committed:
Small docs readme update (#14062)

* Add instruction for running docs locally

* Fix RST syntax

* Update docs/README.rst

Co-authored-by: Kaxil Naik <kaxilnaik@gmail.com>

Co-authored-by: Kaxil Naik <kaxilnaik@gmail.com>

jedcunningham committed:
Only show User's local timezone if it's not UTC (#13904)

ashb committed:
Fix Kerberos network creation on older docker-compose (#14070)

`attachable` is only a property of compose version 3.1 files, but we are
on 2.2 still.

This was failing on self-hosted runners with an error
`networks.example.com value Additional properties are not allowed
('attachable' was unexpected)`

ashb committed:
Fix doc lint error on main (#14093)

ashb committed:
Speed up building provider packages by ~4 times (#14084)

These changes speeds up the build packages step from about 7mins30 to
2min10 on my laptop, and the generate documentation from 3mins to
1min30. On CI it will be even quicker

The changes are:

- Don't do a git fetch for _every_ provider. Once is enough.
- Don't shell out to black, but use the Python API. (Spawning a python
  process is not free)
- As a tidy up, don't print the info about version suffix once per
  package, just once is enough as it doesn't change from package to
  package

Commit:
Make helm chart commands compatible with 1.10.14 image (#13526)

* Make chart command compatible with 1.10.14 image
* ["airflow", "webserver"] does not work with the apache image
* ["webserver"] works, as does "bash" "-c" etc

Co-authored-by: Daniel Standish <dstandish@users.noreply.github.com>
Co-authored-by: Ash Berlin-Taylor <ash_github@firemirror.com>

pchtsp committed:
Fixes missing reference in docs (#14079)

closes: #14045

kaxil committed:
Fix Commands to install Airflow in docker/install_airflow.sh (#14099)

It was trying to run the following command and failing

```
pip install apache-airflow[amazon,google]2.0.0rc2
```

junnplus committed:
Fix order of failed deps (#14036)

potiuk committed:
Corrections in docs and tools after releasing provider RCs (#14082)

kaxil committed:
Update Airflow Releasing Guide with 1.10.x and 2.0 commands (#14098)

Some of the commands were only for 1.10.x

TobKed committed:
Improve GCSToSFTPOperator paths handling (#11284)

* Improve GCSToSFTPOperator paths handling

potiuk committed:
Restores flexible installation version, fixes manual tag build process. (#14107)

Revert "Fix Commands to install Airflow in docker/install_airflow.sh (#14099)"

This reverts commit 68758b826076e93fadecf599108a4d304dd87ac7.

Also fixes the docker build script that was the reason for original
attempt to fix it.

TobKed committed:
Improve Apache Beam operators - refactor operator - common Dataflow logic (#14094)

dimberman committed:
Allow users of the KPO to template environment variables (#14083)

* Allow users of the KPO to template environment variables

Addresses https://github.com/apache/airflow/issues/13348

Adds a "templated_fields" variable to k8s.V1EnvVar so jinja templating
will work when passed by users``

* Update airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py

Co-authored-by: Kaxil Naik <kaxilnaik@gmail.com>

* Update airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py

Co-authored-by: Kaxil Naik <kaxilnaik@gmail.com>

* fix description

Co-authored-by: Kaxil Naik <kaxilnaik@gmail.com>

kaxil committed:
Add a decorator to retry functions with DB transactions (#14109)

This commit adds a decorator: `@retry_db_transaction` to retry
Methods and functions when there is an `OperationalError`. This unifies
the retry configs in a single place and the adds logging.

This makes the code clean and avoids repeatation. This decorator
should not be used when using `@provide_session`.

We already have `run_with_db_retries` to retry code-blocks with the
same default.

`@retry_db_transaction` leverages `run_with_db_retries` (which uses tenacity)
to retry.

You can use this decorator both with and without passing args to it:

`@retry_db_transaction` uses the default configs (example: 3 retries)
If for whatever reason we want to change the number of retries for
a certain function, we can use pass `retries` to decorator,
exmaple: `@retry_db_transaction(retries=2)`

The decorator also accepts kwargs where developers can pass
different keyword arguments to `tenacity.Retrying` object.

potiuk committed:
Change link to keep ex-Polidea employees still INTHEWILD (#14113)

JavierLopezT committed:
Use MongoDB color for MongoToS3Operator (#14103)

Setting UI color for MongoToS3Operator the MongoDB brand one

Co-authored-by: javier.lopez <javier.lopez@promocionesfarma.com>

junnplus committed:
Add statsd extraMappings option to helm chart (#14029)

Commit:
BashOperator to raise AirflowSkipException on exit code 127 (#13421)

potiuk committed:
Remove left-over fields from required in provider_info schema. (#14119)

Those fields were left-over from the full provider.yaml schema.

Apparently they caused no problems but they should be removed.

turbaszek committed:
Add statsd integration to breeze (#12708)

turbaszek committed:
Skip SLA check only if SLA is None (#14064)

Users have to provide SLA as timedelta. If not provided it should
be a default None value.

This change will result in run time errors if other type than
timedelta is passed as SLA. With this change spotting #14056
would be easier.

ashb committed:
Run KinD tests when cncf.kubernetes provider files are changed (#14122)

ashb committed:
Template k8s.V1EnvVar without adding custom attributes to dict. (#14123)

The previous attempt at this worked, but ended up breaking other tests
as the `__eq__` method on this compared to dict.

This customizes the render method to have the variable templated, but
without needing to add an attribute to the V1EnvVar class.

RNHTTR committed:
Add gdrive_to_gcs operator, drive sensor, additional functionality to drive hook, and supporting tests (#13982)

ivica-k committed:
includes the STS token if STS credentials are used (#11227)

as per AWS docs https://docs.aws.amazon.com/redshift/latest/dg/copy-parameters-authorization.html#copy-credentials

turbaszek committed:
Remove duplicated volume mount for git sync in scheduler template (#12717)

closes: #12716

Commit:
Parametrized keda task concurrency in chart (#13571)

* Rely on the config.celery.worker_concurrency value
to determine the number of task a keda worker can take
(vs the previous 16 that was hardcoded in the query).
* Updated documentation accordingly

jlcoto committed:
Fix typos in concept docs (#14130)

brighton1101 committed:
Added support for DSS, ECDSA, and Ed25519 private keys in SSHHook (#12467)

lisovskey committed:
Add allowed_jenkins_states to JenkinsJobTriggerOperator (#14131)

kaxil committed:
Remove reinstalling azure-storage steps from CI / Breeze (#14102)

Since https://github.com/apache/airflow/pull/12188 was merged I
don't think we need this steps.

This step also caused the docker build step for 2.0.1rc2 to fail

Co-authored-by: Jarek Potiuk <jarek@potiuk.com>

Commit:
Fix flower ingress annotations in chart (#13615)

https://github.com/apache/airflow/pull/12010 introduced
a small bug in the way annotations are handled for the
flower ingress.

A test have been added to prevent this from occuring againg.

Also more conditionnals have been added to the
flower-ingress.yaml file to make it compatible with schema
validation.

YingyingPeng22 committed:
#9803 fix bug in copy operation without wildcard  (#13919)

ashb committed:
Run CI (Build Image and committer PRs) on self-hosted runner (#13730)

* Run CI (Build Image and committer PRs) on self-hosted runner

The queue for GitHub actions is starting to get painful.

This PR changes it to run any build of main (push or scheduled), or a PR
from a committer on the self-hosted runner.

However since this setting is configured _in_ the repo, it could be
changed by a PR, so we can't rely on that; we also use a custom build of
the self-hosted runner binary that enforces PRs are from contributors,
otherwise it will fail the build

* Don't free space on self-hosted runners

It deletes the images we have pre-cached (defeating the point of
pre-loading them) and we manage freeing space via another process.

The exception to this is the K8s tests, where we don't need the images
we pre-cache, but we do want the space in our tmpfs.

* Some jobs take longer on self-hosted runners

* When running KinD tests on CI, use pass through the host docker creds to Kind

This enabled self-hosted runners (which are logged in to a docker hub
user) to pull images without running in to pull limits.

* Work around supurious blocking IO failures

This seems to be an occasional problem on the build provider package job
which produces a lot of output all at once.

The error appears as the job just failing with an exit code (sometimes 1, othertimes 120) but no
error.

ashb committed:
Fix typo in Build Images workflow from self-hosted switch (#14150)

The change to this file was untestable until it was merged to master :(

ryanahamilton committed:
Refactor redundant doc url logic to use utility (#14080)

kaxil committed:
Add Airflow 2.0.1 to Changelog and Updating.md (#14151)

Changelog
- https://github.com/apache/airflow/commit/38083fc353befcd012d07e65b29d0c9b9cd0ec7b
- https://github.com/apache/airflow/commit/beb8af5ac6c438c29e2c186145115fb1334a3735

and update the Updating.md

potiuk committed:
Doc fixes after releasing providers. (#14142)

kaxil committed:
Replace 2.0.0 with 2.0.1 in docs (#14153)

mzaforas committed:
Add Paradigma to INTHEWILD.md (#14156)

Paradigma is a technology consultancy company located in Spain that are using airflow in various projects

kaxil committed:
Sync DB Migrations in Master with 2.0.1 (#14155)

`449b4072c2da_increase_size_of_connection_extra_field_.py` does not exist in 2.0.1

so we need to update down_migration

```
INFO  [alembic.runtime.migration] Running upgrade 2c6edca13270 -> 61ec73d9401f, Add description field to connection
INFO  [alembic.runtime.migration] Running upgrade 61ec73d9401f -> 64a7d6477aae, fix description field in connection to be text
INFO  [alembic.runtime.migration] Running upgrade 64a7d6477aae -> e959f08ac86c, Change field in DagCode to MEDIUMTEXT for MySql
INFO  [alembic.runtime.migration] Running upgrade e959f08ac86c -> 82b7c48c147f, Remove can_read permission on config resource for User and Viewer role
[2021-02-09 19:17:31,307] {migration.py:555} INFO - Running upgrade 82b7c48c147f -> 449b4072c2da, Increase size of connection.extra field to handle multiple RSA keys
```

potiuk committed:
Disable progress bar for PIP installation (#14126)

When the image is prepared, PIP installation produces progress
bars which are annoying - especially in the CI environment.

This PR adds argument to control progress bar and sets it to "off"
for CI builds.

Commit:
Fix pod launcher role permissions to list events in chart (#13649)

When using the KubernetesPodOperator with
log_events_on_failure=True, and the pod exits,
the executor needs to access the events in the namespace.

Before this resulted in an exception when using the chart
as this it was not permitted by the pod-launcher role.

jmcarp committed:
Document configuration for email backend credentials. (#14006)

adilkhash committed:
Fix the AttributeError with text field in TelegramOperator (#13990)

closes: #13989

RNHTTR committed:
Fix docs for Google Cloud Secret Manager backend (#14164)

"airflow-tenant-primary" is a bit confusing, and the examples don't match up in the configuration file example.

fix "connections_prefix" and "variables_prefix" mixup in first unordered list in "Managing Secrets" section

make "variable_prefix" plural for consistency

ysktir committed:
Add param to CloudDataTransferServiceOperator (#14118)

When a one-time job is created with `CloudDataTransferServiceS3(GCS)ToGCSOperator`, the job remains on the GCP console even after the job is completed.

This is a specification of the data transfer service, but I would like to add this parameter because there are normally cases where don't want to leave a one-time job.

yuqian90 committed:
Speed up clear_task_instances by doing a single sql delete for TaskReschedule (#14048)

Clearing large number of tasks takes a long time. Most of the time is spent at this line in clear_task_instances (more than 95% time). This slowness sometimes causes the webserver to timeout because the web_server_worker_timeout is hit.

```
        # Clear all reschedules related to the ti to clear
        session.query(TR).filter(
            TR.dag_id == ti.dag_id,
            TR.task_id == ti.task_id,
            TR.execution_date == ti.execution_date,
            TR.try_number == ti.try_number,
        ).delete()
```
This line was very slow because it's deleting TaskReschedule rows in a for loop one by one.

This PR simply changes this code to delete TaskReschedule in a single sql query with a bunch of OR conditions. It's effectively doing the same, but now it's much faster. 

Some profiling showed great speed improvement (something like 40 to 50 times faster) compared to the first iteration. So the overall performance should now be 300 times faster than the original for loop deletion.

SamWheating committed:
Fixes to dataproc operators and hook (#14086)

Two quick fixes to Dataproc operators and hooks.

Add more templated fields to the DataprocClusterDeleteOperator
as per #13454. There were a few other fields which could easily be templated so I added them as well.

Don't use the global-dataproc.googleapis.com:443 URL when creating dataproc clients.
This was partially done in #12907 but the other two client creation methods were not updated. Using the global-dataproc URL results in 404s when trying to create clusters in the global region.

We don't need to specify the default endpoint as it is used by default in the dataproc client library:

https://github.com/googleapis/python-dataproc/blob/6f27109faf03dd13f25294e57960f0d9e1a9fa27/google/cloud/dataproc_v1beta2/services/cluster_controller/client.py#L117

potiuk committed:
Tiny doc fixes after releasing backports (#14173)

kaxil committed:
Log migrations info in consistent way (#14158)

same as #13458 but for `82b7c48c147f_remove_can_read_permission_on_config_.py` migration

This migration changes logging handlers
so each next migration is differently formatted when doing
airflow db reset. This commit fixes this behavior.

ysktir committed:
Add BigQueryUpdateTableOperator (#14149)

jonasrla committed:
Adds Estratégia Educacional to list of Airflow Users (#14180)

jyotidhiman0610 committed:
Add DAG Timeout in UI page "DAG Details" (#14165)

kaxil committed:
Use `Lax` for `cookie_samesite` when empty string is passed (#14183)

closes https://github.com/apache/airflow/issues/13971

The value of `[webserver] cookie_samesite` was changed to `Lax` in >=2.0
from `''` (empty string) in 1.10.x.

This causes the following error for users migrating from 1.10.x to 2.0
if the old airflow.cfg already exists.

```
Traceback (most recent call last):
File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 2447, in wsgi_app
response = self.full_dispatch_request()
File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 1953, in full_dispatch_request
return self.finalize_request(rv)
File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 1970, in finalize_request
response = self.process_response(response)
File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 2269, in process_response
self.session_interface.save_session(self, ctx.session, response)
File "/usr/local/lib/python3.9/site-packages/flask/sessions.py", line 379, in save_session
response.set_cookie(
File "/usr/local/lib/python3.9/site-packages/werkzeug/wrappers/base_response.py", line 468, in set_cookie
dump_cookie(
File "/usr/local/lib/python3.9/site-packages/werkzeug/http.py", line 1217, in dump_cookie
raise ValueError("SameSite must be 'Strict', 'Lax', or 'None'.")
ValueError: SameSite must be 'Strict', 'Lax', or 'None'.**
```

This commit takes care of it by using `Lax` when the value is empty string (``)

Khrol committed:
Use apache.spark provider without kubernetes (#14187)

Catches Name Error on importing optional kubernetes package.

yjwong committed:
Rework client-side script for connection form. (#14052)

* Rework client-side script for connection form.

Before this, fields on the connection form were dynamically changed by
adding and removing the "hide" CSS class. If a provider were to add a
field that requires validation (e.g. InputRequired), or if a different
field type was used (e.g. NumberField), the entire form would be
unsaveable, even if the currently selected connection type had nothing
to do with that provider.

This change takes a different approach; upon loading, the DOM elements
for all extra fields are saved into a Map, then removed from the DOM
tree. When another connection type is selected, these elements are
restored into the DOM.

* Use template string literals in connection form.

Co-authored-by: Ryan Hamilton <ryan@ryanahamilton.com>

Co-authored-by: Ryan Hamilton <ryan@ryanahamilton.com>

Gabriel39 committed:
Update documents for using MySQL (#14174)

Co-authored-by: Jarek Potiuk <jarek@potiuk.com>
Co-authored-by: Kamil Breguła <mik-laj@users.noreply.github.com>

AratiNagmal committed:
Allow AWS Operator RedshiftToS3Transfer To Run a Custom Query (#14177)

Co-authored-by: Arati Nagmal <arati@stepfunction.ai>

Commit:
Add SubprocessHook for running commands from operators (#13423)

This extracts the "run this command" logic from the BashOperator in to a
reusable hook.

Co-authored-by: Daniel Standish <dstandish@users.noreply.github.com>
Co-authored-by: Tomek Urbaszek <turbaszek@gmail.com>
Co-authored-by: Ash Berlin-Taylor <ash_github@firemirror.com>

Jake-Gillberg committed:
Remove accidental cuss (#14207)

bbovenzi committed:
Update Tree View date ticks (#14141)

morrme committed:
Correct typo in GCSObjectsWtihPrefixExistenceSensor  (#14179)

Rename GCSObjectsWtihPrefixExistenceSensor to GCSObjectsWithPrefixExistenceSensor

ryanyuan committed:
Add materialized view support for BigQuery (#14201)

RNHTTR committed:
Reorder doc/spelling build order and improve spelling error message for CI (#14196)

Generic Sphinx errors that were unrelated to spelling were being raised as spelling errors leading to potential confusion. Reversing the order of the build (i.e. running the docs build before the spelling build) will catch such errors and more appropriately denote the issue as a build issue and not a spelling issue. Additionally, the error message for Sphinx errors unrelated to spelling has been updated to be more clear for such cases.

closes: #14051

kaxil committed:
Restructure COMMITTERS.rst (#14218)

- Move "Guidelines to become an Airflow Committer" before "Guidelines for promoting Committers to Airflow PMC"
- Remove "Becoming a Committer" section which is same as "Guidelines to become an Airflow Committer"

ysktir committed:
Add `exists_ok` flag to BigQueryCreateEmptyTable(Dataset)Operator (#14026)

Co-authored-by: uma6 <>
Co-authored-by: Ephraim Anierobi <splendidzigy24@gmail.com>

potiuk committed:
Add better description and guidance in case of sqlite version mismatch (#14209)

Closes: #14208

potiuk committed:
Fixes regexp in entrypoint to include password-less entries (#14221)

Fixes #14104

totogo committed:
Correct a typo in upgrading-to-2.rst (#14225)

Update `list_dag` to `list_dags`

potiuk committed:
Limits Sphinx to <3.5.0 (#14238)

Sphinx 3.5.0 released on 14th of Feb introduced a problem in our
doc builds.

It is documented in https://github.com/sphinx-doc/sphinx/issues/8880

Until this problem is solved we are limiting Sphinx.

Abdur-rahmaanJ committed:
CONTRIBUTING.rst minor link fix and spelling (#14237)

Abdur-rahmaanJ committed:
CONTRIBUTING.rst Link fix from md to rst (#14235)

millin committed:
Fix comparison dagTZ with localTZ (#14204)

potiuk committed:
Disables self-hosted runs for non-apache-owned repositories (#14239)

When the user wants to test pull requests or - especilly - master
build in their own fork, they will not likely have self-hosted
runners configured and the builds will fail.

This change uses self-hosted runners only if the build is run in
the context of the `apache/airflow` repository.

potiuk committed:
fixup! Disables self-hosted runs for non-apache-owned repositories (#14239) (#14242)

Abdur-rahmaanJ committed:
CONTRIBUTING.rst: Clarifying setup step 2 concerning Docker (#14246)

Abdur-rahmaanJ committed:
docs: TESTING.rst: fix not loading image (#14247)

[This image loads](https://github.com/apache/airflow/blob/master/images/testing/run-test.png) but [this](https://github.com/apache/airflow/blob/master/images/testing/run-tests.png) as referred to in the docs, not.

Abdur-rahmaanJ committed:
docs: NOTICE: Updated 2016-2019 to 2016-now (#14248)

Abdur-rahmaanJ committed:
docs: CI.rst: Typos and Link Fix (#14250)

Scuall1992 committed:
Add GoogleDriveToLocalOperator (#14191)

Add new operator to download file from Google Drive to local filesystem. 

Co-authored-by: Tomek Urbaszek <turbaszek@gmail.com>

turbaszek committed:
Add CODEOWNERS for automated PR review assignment (#14216)


Co-authored-by: Kaxil Naik <kaxilnaik@gmail.com>
Co-authored-by: Xiaodong DENG <xd.deng.r@gmail.com>
Co-authored-by: Jarek Potiuk <jarek@potiuk.com>
Co-authored-by: Ash Berlin-Taylor <ash_github@firemirror.com>

HansBambel committed:
Correct PostgreSQL password in doc example code (#14256)

The example code block for PostgreSQL sets the password to the ``airflow_user`` instead of ``airflow_pass``.

zkan committed:
Fix indentation in code block in Taskflow API doc (#14241)

bbovenzi committed:
Js linting and inline migration for simple scripts (#14215)

* Js linting and inline migration for simple scripts

- Linting for all the relatively simple JS in our html templates
- Replaced jQuery .click() and .ready() functions
- changed all js files to snake_case instead of kebab-case for consistency

* add function to get metadata

* fix meta_head and clean up js

* remove extra <head> tag

* snake case task instances file

levahim committed:
Improved unit tests for open_maybe_zipped function. (#14114)

Implemented "real" tests for open_maybe_zipped that test file content
reading without relying on mocks.

pcandoalmeida committed:
Add Airflow UI instance_name configuration option (#10162)

Co-authored-by: Ash Berlin-Taylor <ash@apache.org>

ConstantinoSchillebeeckx committed:
MySQL hook respects conn_name_attr (#14240)

The MySQL hook does not properly use the class attribute `conn_name_attr` in
the get_conn method; this adjusts that method to use the same approach as the
get_uri method.

ashb committed:
Temporarily disable the check-actions step (#14271)

We are now getting "Error: Resource not accessible by integration"
output when we try to run this, even from a triggered workflow should
have the write permissions needed in the token.

To unblock CI I am temporarily disabling this step -- it is not required
(though very nice) as it just creates links from the CI job back to the
build image workflow

jedcunningham committed:
Add more flexibility with FAB menu links (#13903)

Airflow can be more flexible with the links plugins are allowed to add. Currently, you cannot add a top level link, a link with a label, or even without providing a category_icon (which isn't used anyways).

This PR gives plugin authors the flexibility to add any link FAB supports.

ashb committed:
Revert "Temporarily disable the check-actions step (#14271)" (#14284)

This reverts commit 8b5c4a742d42af2410e1fc999b2a295251aa0e6e.

ashb committed:
Fix race in test_celery_executor.py test_retry_on_error_sending_task test (#14273)

The problem was that _sometimes_ the task would actually complete, and
not timeout like we wanted.

This changes the test to _always_ raise a timeout error, instead of
relying on any particular time offset.

Khrol committed:
Don't allow SlackHook.call method accept *args (#14289)

Don't allow SlackHook.call accept *args as underlying Slack WebClient
method doesn't accept *args anymore.

orige committed:
Fix get_context_data doctest import (#14288)

This commit fixes the doctest replacing the old import - from
airflow.task.context - to the newer one.

yogyang committed:
fix lossing duration < 1 secs in tree (#13537)

truncat duration to 3dp when duration < 10

Update airflow/www/views.py

Co-authored-by: Ash Berlin-Taylor <ash_github@firemirror.com>

fix import, retrigger ci

alanwu4321 committed:
fix: Scheduler in helm chart cannot access DAG volume with git sync (#14203)

palerdot committed:
Fix misleading statement on sqlite (#14317)

The statement 
```
By default, Airflow uses **SQLite**, which is *not* intended for development purposes only.
```
is confusing. If `postgres/mysql` are production worthy db backends, and `sqlite` as default db for `airflow` is for development purposes only, this statement is not correct. If I'm mistaken and `sqlite` is for both `production` and `development` purposes, please ignore this PR

junnplus committed:
Helm Chart: Add option to enable/disable flower (#14011)

davidcaron committed:
Fix render_template method name in plugin example (#14278)

these methods exist:

```
flask_appbuilder.BaseView.render_template
flask_admin.BaseView.render
```

this one does not:
```
flask_appbuilder.BaseView.render
```

ecerulm committed:
Add tests for gitSync in helm chart (#14316)

Added Missing tests for #14203

vikramkoka committed:
Added Issue reporting and triage process document (#14223)

* Added Issue reporting and triage process document
Added Issue reporting and triage process document, including the
updated set of labels to be used for reporting and tracking issues.


Based on PR feedback, moved the reference to the issue process doc from the
airflow docs to Contributing doc. Also moved the .rst file to the right directory
and renamed it to match convention.

jedcunningham committed:
Allow viewers to see all docs links (#14197)

Currently, Viewers can only see the "Documentation" link in the docs menu. This PR allows Viewers (and above) to see all of the links.

vikramkoka committed:
Fix link in CONTRIBUTING.rst (#14325)

Fixed a link error in the prior version of the document.

kaxil committed:
Scheduler should not fail when invalid executor_config is passed (#14323)

closes #14182

dave-philp committed:
Added Intellischool (#14330)

Added Intellischool to the official list of Airflow users.

jbampton committed:
Fix various typos (#14335)

jbampton committed:
Remove redundant parentheses from Python file (#14336)

mtahakhan committed:
Docs: Corrected code example description (#14339)

Changed the sentence:
`Here is an example of creating and running a pipeline in Java with jar stored on GCS:`
to 
`Here is an example of creating and running a pipeline in Java with jar stored on local file system:`

potiuk committed:
Fixes failing test_views tests. (#14341)

The get_safe_url previously apparently did not correctly
encode & into %38. I am not sure what was the reason,
(python library fix ??)

But in order to unblock master, this should be fixed.

potiuk committed:
Attempts to stabilize and improve speed of static checks (#14332)

This change attempts to stabilize pylint checks. Since we have
recently added self-hosted runners with multiple CPUS, seems that
re-enabling parallel mode for pylint makes perfect sense as we will
finally be able to use the parallelism and speed up static checks
significantly.

Previously the tests were run in single-processor mode in attempt
to avoid random mistakes where different files were processed in
different processes. This led to random pylint or mypy problems
and aither false-positives or false negatives before especially
when it came to circular dependencies. but since we are now past
heavy refactoring, this should be no problem for future changes
and occasional false positive/negative is less disruptive than
long checks.

The attempt is made to apply sorting order in the files passed
to pylint. This should provide more stability in the results
of running the tests in PR and in master.

We had some custom pylint plugins that prevented using of pylint
parallelism. For now we are giving up on one of the plugins
(no asserts use) and we rely on committer's review on that (we
have a rule in place to only use asserts in tests). The other
plugin was replaced by coming back to separation of "main code"
and "test code" and applying different rules to those - we have
now two different configuration files|: pylintrc and
pylintrc-tests to control settings for those two different cases.

Mypy and flake8 have been parallelized at the level of pre-commits.

By implementing those changes we are likely to speed up the
tests on self-hosted runners 6x-8x times.

Also caching of pre-commit environments (including the
pre-commit installation) is improved. Previously we only cached the
environment created by the pre-commit, but with this change we
also cache the pre-commit installation in .local directory - this
should save 1-2 minutes of the job.

Scuall1992 committed:
Refactor GoogleDriveToGCSOperator to use common methods (#14276)

Refactor GoogleDriveToGCSOperator to use common methods implemented
in hooks used by this operator.

potiuk committed:
Implements generation of separate constraints for core and providers (#14227)

There are two types of constraints now:

* default constraints that contain all depenedncies of airflow,
  all the provider packages released at the time of the relese
  of that version, as well as all transitive dependencies. Following
  those constraints, you can be sure Airflow's installation is
  repeatable

* no-providers constraints - containing only the dependencies needed
  for core airflow installation. This allows to install/upgrade
  airflow without also forcing the provider's to be installed at
  specific version of Airflow.

This allows for flexible management of Airflow and Provider
packages separately. Documentation about it has been added.

Also the provider 'extras' for apache airflow do not keep direct
dependencies to the packages needed by the provider. Those
dependencies are now transitive only - so 'provider' extras only
depend on 'apache-airflow-provider-EXTRA' package and all
the dependencies are transitive. This will help in the future
to avoid conflicts when installing newer providers using extras.

jbampton committed:
Fix spelling (#14343)

Aakcht committed:
Separate resources parameter for kerberos sidecar (#14342)

potiuk committed:
By default PIP will install all packages in .local folder (#14125)

In order to optimize the Docker image, we use the ~/.local
folder copied from build imge (this gives huge optimisations
regarding the docker image size). So far we instructed the users
to add --user flag manually when installing any packages when they
extend the images, however this has proven to be problematic as
users rarely read the whole documentation and simply try what they
know.

This PR attempts to fix it. `PIP_USER` variable is set to `true`
in the final image, which means that the installation by default
will use ~/.local folder as target. This can be disabled by
unsetting the variable or setting it to `false`.

Also since pylint version has been released to 2.7.0, it fixes
a few pylint versions so that we can update to the latest constraints.

potiuk committed:
Fix some tests failures after pylint fixes (#14350)

junnplus committed:
Pprint default args and wrap (#14345)

jbampton committed:
Fix spelling in "ignorable" (#14348)

junnplus committed:
Split webserver config from configmap (#14353)

ashb committed:
Remove testfixtures module that is only used once (#14318)

This is only used in a single test, everywhere else we use Pytest or
unittest's built in feature

potiuk committed:
Fix caching of python images during builds (#14347)

* Fix caching of python images during builds

After GHCR change, github image caching for python images was
broken. GHCR requires each image that we push to it to be
tagged with "org.opencontainers.image.source" label to point
to the right project repository. Otherwise it will not appear
in the repository images and we will not be able to manage
permissions of the image.

So we'v been extending the python base image with appropriate
label, but it was not in all places necessary. As the result.
the builds in CI were usign different images than the cache
image in DockerHub, which in turn caused full image rebuilds
ALWAYS. By implementing this change we make sure tha the the
"airflow-tainted" python images - including the label.

All images (when using breeze) are now built using such
airflow-tainted image which in turn should give5 up to 5-8
minutes saving on building the images job per each job (CI
and production)

The change implements the following behaviours:

1) When image is built on CI or locally without
   UPGRADE_TO_NEWER_DEPENDENCIES flag and without
   FORCE_PULL_IMAGES flag -t the latest image from the
   GitHub Registry or DockerHub registry is used.

2) When both FORCE_PULL_IMAGES and UPGRADE_TO_NEWER_DEPENDENCY
   are set to something different than "false" - the latest Python
   images are used - they are pulled first and tagged appropriately.

This way - always when UPGRADE_TO_NEWER_DEPENDENCIES are set (also
when master pushes are done) - such builds will use latest version
of python base images published on DockerHub.

* Update _push_pull_remove_images.sh

ashb committed:
Remove duplicated clean_db function for ImportErrors (#14296)

There were two functions -- clean_db_errors and clean_db_import_errors
that did the same thing.

cris-b committed:
Fix permission error on non-POSIX filesystem (#13121)

* Fix https://github.com/apache/airflow/issues/12669
 - failure to change ownership of log file on azure

dave-philp committed:
Fixed URL typo for Intellischool (#14372)

URL for Intellischool was mistyped... silly mistake, apologies to the repo mods!

potiuk committed:
Easy switching between GitHub Container Registries (#14120)

This change enables easy switching between GitHub Package Registry
and GitHub Container Registry by simply adding GITHUB_REGISTRY
secret to be either `docker.package.github.com` or `ghcr.io`.

This makes it easy to switch only by the Apache Airflow repository
run builds, as it requires preparation of images (to make them
public and to add permissions to manage them) after they got
created for the first time. GitHub Package Registry works
out-of-the-box but it is less stable and considered a legacy,
also it does not allow image retention.

Documentation has been updated to reflect the reasoning of choosing
this solution as well as describing maintenance processes around
images (including adding new Python version)

Commit:
Fix grammar in production-deployment.rst (#14386)

andygarfield committed:
Fix typo in docker.rst (#14389)

rachael-ds committed:
Documentation and example dag for CloudDLPDeidentifyContentOperator, GCSObjectExistenceSensor, GCSObjectsWithPrefixExistenceSensor (#14033)

* Add documentation and example dag for: CloudDLPDeidentifyContentOperator, GCSObjectExistenceSensor, GCSObjectsWtihPrefixExistenceSensor

* Moving gcs sensor docs and example dags to gcs operators docs/example dags

* Add system tests for dlp and gcs

* Adding further information on DLPDeidentifyContent operators

* Pre-Commit tidyup: Renamed gcs/dlp system tests

* Apply suggestions from code review

Co-authored-by: Kamil Breguła <mik-laj@users.noreply.github.com>

* reverting some changes following code review

* removed redundant @pytest.mark.system("google.cloud")

* removed operators with newly added examples from missing examples list (pytest fix)

* updated all references to GCSObjectsWtihPrefixExistenceSensor (typo) to newly fixed: GCSObjectsWithPrefixExistenceSensor

* fixing merge issue: including deprecated operator to be excluded from test suite of operators

Co-authored-by: rachael-ds <rachael_ds@outlook.com>
Co-authored-by: Kamil Breguła <mik-laj@users.noreply.github.com>

kaxil committed:
Add Ephraim to Committers List (#14397)

https://lists.apache.org/thread.html/r4d6f50e0790c02a28da41066b97bf7b46ddda99bc2949de08647b2df%40%3Cdev.airflow.apache.org%3E

kaxil committed:
Add Ephraim as a Code Owner for the Stable REST API (#14401)

Ephraim already does most of the API reviews so would be good to add him to it

jpyen committed:
Unique pod name (#14186)

* Allow pod name override

* Allow pod name override

kaxil committed:
Add Cross reference for Accessing Task Context in TaskFlow API (#14405)

Lot of users have been asking this on Slack, SO etc on how to do
that. While we have it in our doc, it isn't cross-referenced in
TaskFlow API which causes that feature to be hidden.

kaxil committed:
Sort Committers via their names instead of usernames (#14403)

Previously:

GithubUsername (Name)

Now:

Name (GithubUsername)

JavierLTPromofarma committed:
rebase

JavierLTPromofarma committed:
Merge remote-tracking branch 'origin/snowflake_to_s3' into snowflake_to_s3

# Conflicts:
#	CONTRIBUTING.rst
